### PR TITLE
Support new parameters in detector and rule model.

### PR DESCRIPTION
**Pull request by rahulravindran0108**
SignalFx supports new parameters in their rules and detector
model. These parameters help in defining custom alert messages
and also support adding tags and team-ids to detector.

https://developers.signalfx.com/v2/reference#detector-model

**Testing**
`make test` passes.

### CODE REVIEW

**Comment by kokje** [comment]
If we want terraform to be notified of these changes we need to add it to the resource rule hash, please keep the sorting order in mind so we have a consistent hash value. It would be great if you could update the unit tests as well.

Also, we need to read teams and tags resource properties and add them to the payload we send to SignalFx.

**Comment by rahulravindran0108** [comment]
@kokje I have included the new parameters in the computation of resource hash. Added another test case and ensured that hash value for the previous test case remains consistent. I am guessing this will ensure that terraform does not update all the resources after this change goes through ?

Further, included teams and tags to the payload.

**Comment by poros** [comment]
Can you please update the documentation accordingly. At a first glance, the change looks fine.

File: src/terraform-provider-signalform/signalform/detector.go [hunk 1]
```diff
@@ -173,6 +203,14 @@ func getPayloadDetector(d *schema.ResourceData) ([]byte, error) {
 		payload["visualizationOptions"] = viz
 	}
 
+	if val, ok := d.GetOk("teams"); ok {
+		payload["teams"] = val
```
Comment (kokje) [inline src/terraform-provider-signalform/signalform/detector.go]: In general lgtm! We should specify a type here

**Comment by kokje** [comment]
:shipit:

**Comment by poros** [comment]
Closes #4

**Pull request merged**